likecoin/js/admin/likecoin_metabox.js

```diff
@@ -36,55 +64,212 @@ async function onISCNCallback(event) {
   if (event.origin !== 'https://like.co') {
     return;
   }
+  lcPostInfo.mainStatusTitle = 'READY to #DePub';
+  lcPostInfo.arweaveAndISCNUploadStatus = 'uploading to ISCN...';
+  const arweaveTextField = document.querySelector('#lcArweaveStatus');
   const { action, data } = JSON.parse(event.data);
   if (action !== 'ISCN_SUBMITTED') return;
-  const { tx_hash: txHash, error, success } = data;
+  const {
+    tx_hash: txHash, error, success, iscnId,
+  } = data;
   if (error || success === false) return;
-  const ISCNTextField = document.querySelector('#lcISCNStatus');
-  const res = await jQuery.ajax({
-    type: 'POST',
-    url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/publish/iscn`,
-    dataType: 'json',
-    contentType: 'application/json; charset=UTF-8',
-    data: JSON.stringify({ iscnHash: txHash }),
-    method: 'POST',
-    beforeSend: (xhr) => {
-      xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
-    },
-  });
-  const { url, status } = res;
-  ISCNTextField.innerHTML = `<a rel="noopener" target="_blank" href="${url}">${status}</a>`;
+  try {
+    const res = await jQuery.ajax({
+      type: 'POST',
+      url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/publish/iscn`,
+      dataType: 'json',
+      contentType: 'application/json; charset=UTF-8',
+      data: JSON.stringify({ iscnHash: txHash, iscnId }),
+      method: 'POST',
+      beforeSend: (xhr) => {
+        xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+      },
+    });
+    const { url, status } = res;
+    lcPostInfo.iscnHash = txHash;
+    lcPostInfo.iscnId = iscnId;
+    const iscnIdString = encodeURIComponent(iscnId);
+    arweaveTextField.innerHTML = `<a rel="noopener" target="_blank" href="https://app.rinkeby.like.co/view/${iscnIdString}">${iscnId}</a>`;
+    lcPostInfo.mainStatusTitle = 'LIVE on #DePub';
+    lcPostInfo.arweaveAndISCNUploadStatus = 'success';
+  } catch (err) {
+    console.error(err);
+    const uploadArweaveBtn = document.createElement('button');
+    uploadArweaveBtn.innerText = 'Submit to ISCN';
+    uploadArweaveBtn.setAttribute('id', 'lcArweaveUploadBtn');
+    uploadArweaveBtn.setAttribute('class', 'button button-primary');
+    arweaveTextField.textContent = '';
+    arweaveTextField.appendChild(uploadArweaveBtn);
+    uploadArweaveBtn.addEventListener('click', onEstimateAndUploadArweave);
+    lcPostInfo.mainStatusTitle = 'READY to #DePub';
+    lcPostInfo.arweaveAndISCNUploadStatus = 'failed';
```

Comment:
A very lazy and simple way to handle these post success/fail ui update  is to  just call `onRefreshPublishStatus` after the ajax is done, rather than handle the ui one by one like this

Reply:
Super smart idea! Will do it like this